prog/hints.go

```diff
@@ -22,6 +31,50 @@ func (m ComparisonMapOfSets) AddComp(arg1, arg2 uintptr) {
 	m[arg1][arg2] = true
 }
 
+// Mutates the program using the comparison operands stored in compMaps.
+// For each of the mutants executes the exec callback.
+func (p *Prog) MutateWithHints(compMaps []ComparisonMapOfSets, exec func(newP *Prog)) {
+	for i, c := range p.Calls {
+		if _, ok := hintNamesBlackList[c.Meta.CallName]; ok {
+			continue
+		}
+		foreachArg(c, func(arg, _ Arg, _ *[]Arg) {
+			generateHints(p, compMaps[i], c, arg, exec)
+		})
+	}
+}
+
+func generateHints(p *Prog, compMap ComparisonMapOfSets, c *Call, arg Arg, exec func(newP *Prog)) {
+	candidate := func(arg, newArg Arg) {
```

Comment:
You don't need to pass arg to this function. It already captures the outer function arg, which is the right arg.

Reply:
Still holds.

Reply:
Still holds.